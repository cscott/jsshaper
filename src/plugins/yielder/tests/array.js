var log = (typeof console !== "undefined") && console.log || print;
(typeof require !== 'undefined') && require('setup.js');

for each (i in [1,2,3]) { log(i); }
// array comprehensions
function range(begin, end) {
  for (var i = begin; i < end; ++i) {
    yield i;
  }
}

var ten_squares = [ i * i for each (i in range(0, 10))];
log(ten_squares.join(','));

var evens = [ /*a*/ i for each (i in range(0, 21)) if (i % 2 == 0)];
log(evens.join(','));

/* generator expressions */
function foo(it) {
    return it.next();
}
log(foo(i for each (i in range(1, 5))));
