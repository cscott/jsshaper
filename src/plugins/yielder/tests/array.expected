1
2
3
0,1,4,9,16,25,36,49,64,81
0,2,4,6,8,10,12,14,16,18,20
1
